log-collector/src/main/scala/io/phdata/pulse/logcollector/SolrCloudStream.scala

```diff
@@ -0,0 +1,176 @@
+  logger.info(s"Overflow stragegy `pulse.collector.stream.overflow.strategy` ${overflowStrategy}")
+  logger.info(s"Batch size `pulse.collector.stream.batch.size`is $solrBatchSize")
+  logger.info(s"Batch flush duration `pulse.collector.stream.flush.seconds` is $batchFlushDuration")
+  logger.info(
+    s"Number of threads used to publish to Solr `pulse.collector.stream.numthreads` is $numThreads")
+
+  // This stratefy will fail by signalling a function that exits the application. The original `OverflowStrategy.Fail`
+  // doesn't have a clear way get a stream failure message.
+  private val failOverflowStrategy =
+    OverflowStrategy.ClearBufferAndSignal(maxBuffersize, exitingOverflowMessage _)
+
+  private val overflowStrategy =
+    sys.props
+      .get("pulse.collector.stream.overflow.strategy")
+      .map { strategy =>
+        strategy.trim.toLowerCase match {
+          case "fail" =>
+            failOverflowStrategy
+          case "dropold" =>
+            OverflowStrategy.DropOldAndSignal(maxBuffersize, overflowMessage _)
+          case "dropnew" =>
+            OverflowStrategy.DropNewAndSignal(maxBuffersize, overflowMessage _)
+          case "clearbuffer" =>
+            OverflowStrategy.ClearBufferAndSignal(maxBuffersize, overflowMessage _)
+          case "backpressure" =>
+            OverflowStrategy.BackPressure(maxBuffersize)
+          case _ => throw new IllegalArgumentException(s"Unrecogized overflow strategy $strategy")
+        }
+      }
+      .getOrElse {
+        failOverflowStrategy
+      }
+
+  // Scheduler handles blocking http calls to Solr
+  private val blockingScheduler = {
+    val javaService = Executors.newScheduledThreadPool(numThreads)
+    Scheduler(javaService, AlwaysAsyncExecution)
+  }
+
+  // Create a subject we can write events to
+  private val subject = ConcurrentSubject
+    .publish[(String, Map[String, String])](overflowStrategy.asInstanceOf[Synchronous[Nothing]])
+
+  /* Perform the transformation:
+     - group the events by their application name, creating multiple streams, one for each application
+     - group the events in batches of [[solrBatchSize]] or after [[solrFlushDuration]]
+     - write the groups to solr
+     - subscribe to the stream so every event added to the stream runs this workflow
+   */
+  private val transformation =
+    subject
+      .groupBy(x => x._1)
+      .map { x =>
+        x.bufferTimedAndCounted(batchFlushDuration, solrBatchSize)
+          .mapAsync(y => save(x.key, y.map(_._2)))
+          .executeOn(blockingScheduler)
+          .foreach(x => ())
+      }
+      .subscribe()
+
+  /**
+   * Put an event onto the stream
+   *
+   * @param applicationName Name of the application
+   * @param data            The data object, key value pairs
+   */
+  private[logcollector] def put(applicationName: String, data: Map[String, String]): Unit =
+    subject.onNext((applicationName, data)).onComplete(x => x.isSuccess)
```

Comment:
What's the purpose of the `.onComplete` here?  It doesn't appear to return a value or log info.